### PR TITLE
PUBDEV-7138 - Extended Isolation Forest

### PR DESCRIPTION
**Pull request by maurever**
PR to run test on jenkins.

Author: @valenad1

### CODE REVIEW

**Comment by maurever** [comment]
@hannah-tillman could you please check the documentation parts of this PR? The Extended Isolation Forest is a completely new algorithm in h2o. What else is necessary to document here? Let me know if you have any questions. Thank you!

File: h2o-docs/src/product/data-science/eif.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Tutorials and Blogs
+~~~~~~~~~~~~~~~~~~~
+
+The following tutorials are available that describe how to use Extended Isolation Forest. 
+
+- `Master's thesis: Anomaly detection using Extended Isolation Forest <https://dspace.cvut.cz/bitstream/handle/10467/87988/F8-DP-2020-Valenta-Adam-thesis.pdf?sequence=-1&isAllowed=y>`__: The thesis deals with anomaly detection algorithms with a focus on the Extended Isolation Forest algorithm including the implementation to the H2O-3 open-source Machine Learning platform.
+- `Extended Isolation Forest jupyter notebook created by the authors of the algorithm <https://github.com/sahandha/eif/blob/master/Notebooks/EIF.ipynb>`__: Describes how Extended Isolation Forest behaves compare to Isolation Forest.
+
+
+Defining an Extended Isolation Forest Model
```
Comment (hannah-tillman) [inline h2o-docs/src/product/data-science/eif.rst:39]: I've noticed that there a several parameters that are listed in the H2OExtendedIsolationForestEstimator python file that aren't listed in the eif.rst file: `col_sample_rate_change_per_level`, `col_sample_rate_per_tree`, `export_checkpoints_dir`, `max_depth`, `min_rows`, `score_each_iteration`, `score_tree_interval`, `stopping_metric`, `stopping_rounds`, & `stopping_tolerance`. Is there a reason for that?

File: h2o-py/h2o/estimators/extended_isolation_forest.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+        """
+        Whether to score during each iteration of model training.
+
+        Type: ``bool``  (default: ``False``).
+
+        :examples:
+
+        >>> cars = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/junit/cars_20mpg.csv")
+        >>> predictors = ["displacement","power","weight","acceleration","year"]
+        >>> cars_if = H2OExtendedIsolationForestEstimator(score_each_iteration = True,
```
Comment (hannah-tillman) [inline h2o-py/h2o/estimators/extended_isolation_forest.py:80]: There's a lot of line mismatch? I don't know how to explain it--where the next parameter listed in building the estimator isn't directly beneath the line above it (e.g. how `ntrees` isn't directly below `score_each_iteration`). 

Would you like me to go through and fix this as well?

File: h2o-docs/src/product/data-science/eif.rst (hunk 1 above)
Comment (maurever) [reply h2o-docs/src/product/data-science/eif.rst:39]: Definitely a bug, could you @valenad1 take a look, please?

File: h2o-docs/src/product/data-science/eif.rst [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+The general case needs to define a random slope for each branching point.
+Instead of selecting the feature and value, it selects a random slope :math:`n` for
+the branching cut and a random intercept :math:`p`. The slope can
+be generated from :math:`\mathcal{N(0,1)` Gaussian distribution, and the
+intercept is generated from the uniform distribution with bounds coming
+from the sub-sample of data to be split. The branching criteria for the data
+splitting for a given point :math:`x` is as follows:
+
+.. math::
+    (x - p) � n < 0
```
[the remark this answers is not in the document]
Comment (maurever) [reply h2o-docs/src/product/data-science/eif.rst]: I think it should be *. @valenad1 could you approve, please?

File: h2o-py/h2o/estimators/extended_isolation_forest.py (hunk 2 above)
Comment (maurever) [reply h2o-py/h2o/estimators/extended_isolation_forest.py:80]: Maybe I dont understand. If the parameters setter and getter are in different order, I think it is ok. :)

File: h2o-docs/src/product/data-science/eif.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (valenad1) [reply h2o-docs/src/product/data-science/eif.rst]: Approve

File: h2o-docs/src/product/data-science/algo-params/extension_level.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+For the dataset with :math:`P` features the maximum extension_level is :math:`P-1` and means full-extension. 
+As the extension_level is increased the bias of standard Isolation Forest is reduced. A lower extension is suitable for
+a domain where the range of the minimum and maximum for each feature highly differs (for example, when one feature is 
+measured in millimeters, and the second one in meters). For the more detailed explanation serves the following paragraphs.
+
+The branching criteria in Extended Isolation Forest for the data
+splitting for a given data point :math:`x` is as follows:
+
+.. math::
+    (x - p) � n < 0
```
[the remark this answers is not in the document]
Comment (valenad1) [reply h2o-docs/src/product/data-science/algo-params/extension_level.rst]: It is ok.

File: h2o-docs/src/product/data-science/eif.rst (hunk 1 above)
Comment (valenad1) [reply h2o-docs/src/product/data-science/eif.rst:39]: Fixed in: ff3dcb86a04ea08f4b6373fc1dcf2abcd751f13e

File: h2o-py/h2o/estimators/extended_isolation_forest.py (hunk 2 above)
Comment (valenad1) [reply h2o-py/h2o/estimators/extended_isolation_forest.py]: I understand, I will fix the formating of model attributes

Comment (valenad1) [reply h2o-py/h2o/estimators/extended_isolation_forest.py]: Fixex

**Comment by valenad1** [comment]
The purpose of this PR only to run test is fulfilled.

The production version of algorithm will be finished here: https://github.com/h2oai/h2o-3/pull/5246

**Pull request closed without merging**